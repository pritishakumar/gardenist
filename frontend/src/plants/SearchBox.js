import React, { useState } from 'react';
// Material UI Components
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import FilledInput from '@material-ui/core/FilledInput';
import FormHelperText from '@material-ui/core/FormHelperText';
import Button from '@material-ui/core/Button';

/** Component rendered in ./Search Component. It is
 * the form element which the user types their search
 * query into it. Component passes the final submitted
 * query value up to parent.
 * 
 * Authorization: none, accessible to all
 */
function SearchBox({ setFxn, query }) {
	const [ formData, setFormData ] = useState(query);

	async function handleSubmit(evt) {
		evt.preventDefault();
		await setFxn(formData);
		setFormData('');
	}

	return (
		<form onSubmit={handleSubmit} class="Search">
			<FormControl variant="filled">
				<InputLabel htmlFor="input-search">Search keywords: </InputLabel>
				<FilledInput
					name="search"
					id="input-search"
					value={formData}
					onChange={(evt) => setFormData(evt.target.value)}
				/>
				{formData === '' || formData.match(/^[a-z\s]+$/i) ? null : (
					<FormHelperText error className="component-error-text">
						Must contain only English letters and spaces!
					</FormHelperText>
				)}
			</FormControl>
			<br />
			<Button variant="contained" color="primary" type="submit">
				Search Plants
			</Button>
		</form>
	);
}

export default SearchBox;
