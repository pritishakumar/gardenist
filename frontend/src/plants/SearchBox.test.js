import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom';
import SearchBox from "./SearchBox";

const fxn = () => {return null}

it("renders without crashing", function() {
  render(
    <MemoryRouter>
        <SearchBox setFxn={fxn} query="rose" />
    </MemoryRouter>
    );
});

it("matches snapshot", function() {
    const { asFragment } = render(
        <MemoryRouter>
            <SearchBox setFxn={fxn} query="rose" />
        </MemoryRouter>
        );
    expect(asFragment()).toMatchSnapshot();
  });