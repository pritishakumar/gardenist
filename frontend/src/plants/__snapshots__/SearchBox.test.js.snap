// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="Search"
  >
    <div
      class="MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
        data-shrink="true"
        for="input-search"
      >
        Search keywords: 
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiFilledInput-input"
          id="input-search"
          name="search"
          type="text"
          value="rose"
        />
      </div>
    </div>
    <br />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Search Plants
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
